pkg/providers/cloudstack/cloudstack.go

```diff
@@ -1169,6 +1173,7 @@ func (p *cloudstackProvider) UpgradeNeeded(ctx context.Context, newSpec, current
 	if err != nil {
 		return false, err
 	}
+	currentSpec.CloudStackDatacenter = existingCsdc
```

Comment:
same comment here about side effect

Reply:
Please refer to https://github.com/aws/eks-anywhere/pull/2915#discussion_r943642635